tests/FSharp.Test.Utilities/CompilerAssert.fs

```diff
@@ -27,6 +27,17 @@ open Xunit
 open TestFramework
 open System.Collections.Immutable
 
+type FileNames =
+    static let testFileName = AsyncLocal<string>()
+    static let mutable counter = 0
+    //static do testFileName.Value <- "test"
+    static member internal MakeTestFileNameUniqueForThisTestCase() =
+        testFileName.Value <- $"{Interlocked.Increment &counter}{Path.PathSeparator}test"
+
```

Comment:
I guess you didn't want to use `Path.PathSeparator`, which is "A platform-specific separator character used to separate path strings in environment variables."